pkg/controller/controller.go

```diff
@@ -72,7 +81,14 @@ func New(name string, informer cache.SharedIndexInformer, startCache func(contex
 	informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
 		AddFunc: controller.handleObject,
 		UpdateFunc: func(old, new interface{}) {
-			controller.handleObject(new)
+			if !opts.SkipNoUpdate {
+				// SkipNoUpdate is disabled, objects will occur regardless of whether an update actually took place.
+				controller.handleObject(new)
+			}
+			// SkipNoUpdate is enabled, objects will only be handled if they have changed.
```

Comment:
Did you want to add either a return or an else in between these two conditions?

Reply:
Yes, good catch but ended up combining this anyway per the suggestion below.